BetterSnippetManager.py

```diff
@@ -98,6 +96,28 @@ def set_description(self, description):
     def set_scopes(self, scopes):
         self.scopes = self.escape(scopes)
         folder = self.scopes.split(' ')[0].split('.')[-1]
+
+        user_folder = get_settings().get('snippets_folder')
+        use_base_folder = get_settings().get('use_sublime_snippets_folder_as_base_folder')
+
+        # use base folder + custom folder
+        if use_base_folder and user_folder:
+            folder = os.path.join("Snippets", user_folder)
+
+        # use base folder only
+        elif use_base_folder and user_folder is None:
+            folder = "Snippets"
+
+        # use base folder + scope folder
+        elif use_base_folder:
+            folder = os.path.join("Snippets", folder)
+
+        # use custom folder
+        elif user_folder:
+            folder = user_folder
+
+        # otherwise leave folder variable unchanged and use scope folder
+
         view = self.window.show_input_panel('Folder: ', folder, self.set_folder, None, None)
```

Comment:
I expanded this section so that you can define better where your snippets will go:

If you set **_"use_sublime_snippets_folder_as_base_folder": true_** you can have these combinations:

| snippets_folder           | Target Dir|  
| --------------------    | ----------------------     |  
| "" | User/Snippets/(generated from scope) |  
| null | User/Snippets |  
| "SomeDir | User/Snippets/Custom |  


If you set **_"use_sublime_snippets_folder_as_base_folder": false_** you can have these combinations:

| snippets_folder           | Target Dir|  
| --------------------    | ----------------------     |  
| ""  or null | User/(generated from scope) |  
| "SomeDir | User/Custom |